baseTrie.js

```diff
@@ -360,7 +360,29 @@ Trie.prototype._updateNode = function (key, value, keyRemainder, stack, cb) {
 
   // add the new nodes
   key = TrieNode.stringToNibbles(key)
-  if (lastNode.type === 'branch') {
+
+  // Check if the last node is a leaf and the key matches to this
+  var matchLeaf = false
+  if (lastNode.type === 'leaf') {
```

Comment:
Not sure, if this is the most efficient way to do the key length comparison, won't take this as a blocker though.